numpyro/distributions/transforms.py

```diff
@@ -1347,26 +1347,25 @@ def __call__(self, x: jnp.ndarray) -> jnp.ndarray:
         x = jnp.moveaxis(x, -2, 0)
 
         def f(y, x):
-            y = (self.transition_matrix * y[..., None, :]).sum(axis=-1) + x
+            y = y @ self.transition_matrix.T + x
```

Comment:
I think you might want `jnp.swapaxes(self.transition_matrix, -1, -2)`

Reply:
I've updated it to use `einsum` because we have shapes `(..., p, p)` for the transition matrices `A` and `(..., n, p)` for the states `x`. Inside the `scan` function, we are dealing with state of shape `(..., p)` because we're scanning along the `n` dimension. There may very well be a better way to do this.